vcd/resource_vcd_independent_disk.go

```diff
@@ -173,33 +197,180 @@ func resourceVcdIndependentDiskCreate(d *schema.ResourceData, meta interface{})
 
 	diskCreateParams.Disk.Description = d.Get("description").(string)
 
+	if value, ok := d.GetOk("sharing_type"); ok {
+		diskCreateParams.Disk.SharingType = value.(string)
+	}
+
 	task, err := vdc.CreateDisk(diskCreateParams)
 	if err != nil {
-		return fmt.Errorf("error creating independent disk: %s", err)
+		return diag.Errorf("error creating independent disk: %s", err)
 	}
 
 	err = task.WaitTaskCompletion()
 	if err != nil {
-		return fmt.Errorf("error waiting to finish creation of independent disk: %s", err)
+		return diag.Errorf("error waiting to finish creation of independent disk: %s", err)
 	}
 
 	diskHref := task.Task.Owner.HREF
 	disk, err := vdc.GetDiskByHref(diskHref)
 	if err != nil {
-		return fmt.Errorf("unable to find disk with href %s: %s", diskHref, err)
+		return diag.Errorf("unable to find disk with href %s: %s", diskHref, err)
 	}
 
 	d.SetId(disk.Disk.Id)
 
-	return resourceVcdIndependentDiskRead(d, meta)
+	return resourceVcdIndependentDiskRead(ctx, d, meta)
+}
+
+func resourceVcdIndependentDiskUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	if d.HasChanges("size_in_mb", "storage_profile", "description") {
```

Comment:
Would `description` change really require to detach/reattach the disks to VMs? It essentially becomes a reboot operation for a cluster of VMs

Reply:
One more time validate that:

The requested operation cannot be performed because the disk \"TestDisk\" is attached to the virtual machine \"Photon OS\".")

Even for Description update

Reply:
Ok - thanks for checking. An "expensive" name change, but that is not up to us.